docs/user_manual/components.md

```diff
@@ -18,7 +18,9 @@ The base type for all power-grid-model components.
 
 | name | data type | unit | description                                                                                                                                                       | required |                                     update                                     |
 | ---- | --------- | ---- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------: | :----------------------------------------------------------------------------: |
-| `id` | `int32_t` | -    | ID of a component. The ID should be unique across all components within the same scenario, e.g., you cannot have a node with `id=5` and another line with `id=5`. | &#10004; | &#10060; (id needs to be specified in the update query, but cannot be changed) |
+| `id` | `int32_t` | -    | ID of a component. The ID should be unique across all components within the same scenario, e.g., you cannot have a node with `id=5` and another line with `id=5`. | &#10004; | &#10060; (If a component update is uniform (see below), the ID can be omitted or set to `nan`. Otherwise, the id needs to be specified in the update query, but cannot be changed) |
+
+A uniform component update is defined as one in which every scenario modifies the same number of elements within that component. Therefore, a dense update is always uniform, hence the ID can always be optional; whereas a sparse update is only uniform if the index pointer is given as a linear sequence of elements (sparse representation of a dense buffer). An example of the usage of optional IDs is given in [Power Flow Example](./Power%20Flow%20Example.ipynb)
```

Comment:
```suggestion
Uniform component updates ones that update the same number of component elements across scenarios. By definition, a dense update is always uniform, hence the ID can always be optional; ...
```

``` a sparse update is only uniform if the index pointer is given as a linear sequence of elements```

I don't think this is necessarily true. Sequence `{1, 2, 3}` is linear, just like `{2, 4, 6}` is linear, and `{6, 4, 2}`.

Reply:
Maybe consider:
``` suggestion
... whereas a sparse update is only uniform if the index pointer is given as an arithmetic sequence of all elements (sparse representation of a dense buffer). An example of the usage of optional IDs is given in [Power Flow Example](./Power%20Flow%20Example.ipynb)
```